gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GcsJavaClientImpl.java

```diff
@@ -182,12 +227,47 @@ public Map<String, Long> getStatistics() {
 
   @Override
   public void close() {
-    gcsClientDelegate.close();
+    try {
+      try {
+        gcsClientDelegate.close();
+      } finally {
```

Comment:
it might be a good idea to catch any potential exceptions thrown and log it.

Reply:
added.